### PR TITLE
fix(carousel): Better a11y with sr-only text and markup structure

### LINKED ISSUE #4478: Default carousel template is not accessible
* Angular - 1.2.28
* ui-bootstrap - 0.11.2

Here is a plunker of a failing scenario: http://plnkr.co/edit/pouPdsf3ugooUiyHMPTV?p=preview

The default template is not accessible via keyboard navigation nor with screen reader assistance. While tabbing through the document you should be able to get keyboard focus on the carousel controls. While using assistive technology, you should also be able to tab-stop on the carousel controls. Neither of these are working as-is.

I understand that newer versions allow for custom templates but there are very minimal changes to the default template that would allow all versions to be much more accessible.

### PR DESCRIPTION
**Pull request by eduncan26**
fixes #4478

### CODE REVIEW

File: template/carousel/carousel.html [hunk 1]
```diff
@@ -1,8 +1,16 @@
 <div ng-mouseenter="pause()" ng-mouseleave="play()" class="carousel" ng-swipe-right="prev()" ng-swipe-left="next()">
-    <ol class="carousel-indicators" ng-show="slides.length > 1">
-        <li ng-repeat="slide in slides | orderBy:indexOfSlide track by $index" ng-class="{active: isActive(slide)}" ng-click="select(slide)"></li>
-    </ol>
-    <div class="carousel-inner" ng-transclude></div>
-    <a class="left carousel-control" ng-click="prev()" ng-show="slides.length > 1"><span class="glyphicon glyphicon-chevron-left"></span></a>
-    <a class="right carousel-control" ng-click="next()" ng-show="slides.length > 1"><span class="glyphicon glyphicon-chevron-right"></span></a>
-</div>
+  <div class="carousel-inner" ng-transclude></div>
+  <a role="button" href="#" class="left carousel-control" ng-click="prev()" ng-show="slides.length > 1">
```
Comment (wesleycho) [inline template/carousel/carousel.html]: This needs to be an empty `href` due to some edge cases

File: template/carousel/carousel.html [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    <span aria-hidden="true" class="glyphicon glyphicon-chevron-left"></span>
+    <span class="sr-only">previous</span>
+  </a>
+  <a role="button" href="#" class="right carousel-control" ng-click="next()" ng-show="slides.length > 1">
```
Comment (wesleycho) [inline template/carousel/carousel.html]: Ditto

**Comment by wesleycho** [comment]
Other than those two small issues that we can fix on merge, I'm fine with this.

**Comment by eduncan26** [comment]
Sweet!

**Pull request closed without merging**